src/datasources/cache/redis.cache.service.ts

```diff
@@ -31,7 +31,7 @@ export class RedisCacheService
     return this.client.ping();
   }
 
-  async set(
+  async setWithExpiration(
```

Comment:
By splitting the parameters of the `SET` operation into multiple functions, we increase the number of functions that we need to implement in case we want to extend its functionality.

If we need to check if a specific component was called with a certain number of parameters, a mocked object might be a better fit for this (while we can still have an integration test with the actual implementation to check if the TTL was set).

See SET available options – https://redis.io/commands/set/

Reply:
The gotcha here is we are not using `SET` but `HSET`, which does not provide these options to manage the TTL. But this is not relevant anymore as I think the PR you merged (https://github.com/safe-global/safe-client-gateway/pull/1219) solves this problem accurately by passing the `NX` option into the `expire` call 🙂

I will reset this PR and use it exclusively to align our `ICacheService` implementations.